core/src/providers/mistral.rs

```diff
@@ -387,30 +400,46 @@ impl MistralAILLM {
             .map(|m| MistralChatMessage::try_from(m))
             .collect::<Result<Vec<_>>>()?;
 
-        // Mistral AI requires a tool call to be followed by a tool message. We therefore inject a
-        // tool call message before the tool message if missing.
+        // Mistral AI requires a tool call assistant message to be followed by a tool message. We
```

Comment:
Once we phase-out single action assistants (and therefore the old renderConversationForModel), should we remove this logic ?

Reply:
We should yes